File: eRents.Domain/Models/AddressDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eRents.Domain.Models;

public partial class AddressDetail
{
    public int AddressDetailId { get; set; }

    public int GeoRegionId { get; set; }

    [Required]
    [StringLength(255)]
    public string StreetLine1 { get; set; } = null!;

    [StringLength(255)]
    public string? StreetLine2 { get; set; }

    [Column(TypeName = "decimal(9, 6)")]
    public decimal? Latitude { get; set; }

    [Column(TypeName = "decimal(9, 6)")]
    public decimal? Longitude { get; set; }

    public virtual GeoRegion GeoRegion { get; set; } = null!;

    public virtual ICollection<Property> Properties { get; set; } = new List<Property>();

    public virtual ICollection<User> Users { get; set; } = new List<User>();
} 

File: e_rents_desktop/lib/models/address_detail.dart
import './geo_region.dart';

class AddressDetail {
  final int addressDetailId;
  final int geoRegionId;
  final String streetLine1;
  final String? streetLine2;
  final double? latitude;
  final double? longitude;
  final GeoRegion? geoRegion; // Nested GeoRegion object

  AddressDetail({
    required this.addressDetailId,
    required this.geoRegionId,
    required this.streetLine1,
    this.streetLine2,
    this.latitude,
    this.longitude,
    this.geoRegion,
  });

  factory AddressDetail.fromJson(Map<String, dynamic> json) {
    return AddressDetail(
      addressDetailId: json['addressDetailId'] as int,
      geoRegionId: json['geoRegionId'] as int,
      streetLine1: json['streetLine1'] as String,
      streetLine2: json['streetLine2'] as String?,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      geoRegion:
          json['geoRegion'] != null
              ? GeoRegion.fromJson(json['geoRegion'] as Map<String, dynamic>)
              : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'addressDetailId': addressDetailId,
      'geoRegionId': geoRegionId,
      'streetLine1': streetLine1,
      'streetLine2': streetLine2,
      'latitude': latitude,
      'longitude': longitude,
      'geoRegion': geoRegion?.toJson(),
    };
  }
}
